src/main/protobuf/build/buildfarm/v1test/buildfarm.proto

```diff
@@ -39,6 +43,42 @@ message StopContainerRequest {
   string container_name = 3;
 }
 
+message GetHostsRequest {
+  string filter = 1;
+
+  int32 age_in_minutes = 2;
+
+  string status = 3;
+}
+
+message GetHostsResult {
+  int64 num_hosts = 1;
+
+  repeated Host hosts = 2;
+}
+
+message Host {
+  int64 host_num = 1;
+
+  string host_id = 2;
+
+  string ip_address = 3;
+
+  google.protobuf.Timestamp launch_time = 4;
+
+  string state = 5;
+
+  string dns_name = 6;
+
+  int64 uptime_minutes = 7;
+
+  int64 num_cores = 8;
+
+  string type = 9;
+
+  string lifecycle = 10;
```

Comment:
this looks like it should be an enum, rather than a freeform string.

Reply:
AWS defines it as a string here. For most use cases this will be a single value or an empty string so I'd rather keep it as a string.